### PR TITLE
Add lockdown setting to gui

### PR DESCRIPTION
**Pull request by mlsteele**
image

label wraps on mobile

### CODE REVIEW

**Comment by maxtaco** [comment]
That's great, I like that wording.  One concern -- does it convey to people that they are royally f'ed if they lose all their devices, that reset isn't a possibility in that case?  That is bound to happen.

File: shared/settings/advanced/index.js [hunk 1]
```diff
@@ -20,9 +22,20 @@ const Advanced = (props: Props) => (
...
       flex: 1,
-      padding: globalMargins.medium,
+      paddingLeft: globalMargins.medium,
+      paddingRight: globalMargins.medium,
+      paddingBottom: globalMargins.medium,
     }}
   >
+    <Box style={{...globalStyles.flexBoxRow, minHeight: 48, alignItems: 'center'}}>
+      <Checkbox
+        checked={props.lockdownModeEnabled || false}
```
Comment (songgao) [inline shared/settings/advanced/index.js]: It looks like this can be `!!props.lockdownModeEnabled`?

**Comment by chrisnojima** [comment]
I don't think this is really necessary. If you lost all your devices you've
lost all your data. The reset just lets them keep their username

File: shared/actions/settings.js [hunk 2]
```diff
@@ -400,6 +400,27 @@ const _rememberPassphraseSaga = (action: SettingsGen.OnChangeRememberPassphraseP
...
+    })
+
+const setLockdownMode = (state: TypedState, action: SettingsGen.OnChangeLockdownModePayload) =>
+  state.config.loggedIn &&
+  RPCTypes.accountSetLockdownModeRpcPromise({enabled: action.payload.enabled}, Constants.waitingKey)
+    .then(() => {
+      return SettingsGen.createLoadedLockdownMode({status: action.payload.enabled})
+    })
+    .catch(() => {
+      return loadLockdownMode(state)
```
Comment (buoyad) [inline shared/actions/settings.js]: can you return the load action instead?

**Comment by mlsteele** [comment]
Feel free to merge while I'm out

**Comment by maxtaco** [comment]
@buoyad did @mlsteele address your comment?

**Comment by buoyad** [comment]
@maxtaco yep, we also talked about changing the actions a bit but that can be done separately if you want to merge

**Comment by maxtaco** [comment]
crudo. merge conflicts

**Comment by maxtaco** [comment]
@zapu I was going to ask if you could push this across the finish line, but you've gone ahead and done it. Great, thank you!

**Comment by zapu** [comment]
Thanks! Just taking a last look and waiting for CI to finish

**Pull request merged**